### PR TITLE
Regenerate helix client and update for new api version

### PR DESCRIPTION
**Pull request by alexperovich**
This change generates code for the new helix api version and updates the JobSender and MSBuild SDK to work with the new apis.

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Client/CSharp/generated-code/WorkItem.cs [hunk 1]
```diff
@@ -43,6 +56,193 @@ public WorkItem(HelixApi client)
...
+            {
+                _req?.Dispose();
+                _res?.Dispose();
+                throw;
+            }
+        }
+
+        partial void HandleFailedListFilesRequest(RestApiException ex);
+
+        public async Task<IImmutableList<UploadedFile>> ListFilesAsync(
```
Comment (mjanecke) [inline src/Microsoft.DotNet.Helix/Client/CSharp/generated-code/WorkItem.cs]: Woot woot!

**Comment by alexperovich** [comment]
AzDev messed up the check. Build is all green, merging.

**Pull request merged**